### ISSUE TITLE
Any plan to support folding batch norm in quantized-aware training?

### ISSUE THREAD

**Issue by Zehaos** [NervanaSystems/distiller#166]
It seems that current implementation do not support folding batch norm in quantized-aware training, any plan to support it?

Page18, Quantizing deep convolutional networks for efficient inference:A whitepaper.

**Comment by guyjacob** [comment]
At the moment we're missing some needed infrastructure to perform proper fusion of layers in general. Batch-norm folding is an example of such fusion. This is something I hope we'll get to very soon. We'll implement that along with batch-norm folding for **post-training quantization**.
The folding scheme for training described in the paper you refer to is a bit more involved and will not be supported at first. Once the infrastructure I mentioned is in-place we will revisit this feature.

**Comment by levzlotnik** [comment]
Hi, 
This was implemented in #274, though not the automatic fusion. You could use `distiller.model_transforms.fold_bn_norms` to pre-fold the batch norms in the network, and it will be ready for QAT.

Closing this issue due to old age, feel free to reopen if needed.
